distributed/scheduler.py

```diff
@@ -2909,6 +2910,13 @@ def worker_objective(self, key, worker):
         start_time = comm_bytes / BANDWIDTH + stack_time
         return (start_time, self.worker_bytes[worker])
 
+    def cause_of_failure(self, stream=None, keys=None):
+        for key in keys:
+            if key in self.exceptions_blame:
+                cause = self.exceptions_blame[key]
+                # cannot serialize sets
+                return (list(self.dependencies[cause]), cause,
+                        self.tasks[cause])
```

Comment:
We might consider sending back a dict, which might be a bit more self-documenting.